### PR TITLE
Commit to new modular architecture

### PR DESCRIPTION
**Pull request by gonuke**
This PR replaces #33 - the code is the same set of commits with one additional commit for the README file.

This branch has been the de facto development branch for over a year.  It is not perfect, but it makes sense to commit to it going forward and then make improvements.

### CODE REVIEW

**Comment by gonuke** [comment]
It would be great to get this reviewed and merged.... I essence this is the only branch we are using anyway, and merging it will just make our documentation a little easier

File: MCNPInput.hpp [hunk 1]
```diff
@@ -3,12 +3,12 @@
 
 #include <vector>
 #include <map>
-#include <iosfwd>
 #include <string>
+#include <cmath>
```
[the remark this answers is not in the document]
Comment (gonuke) [reply MCNPInput.hpp]: Definitely a good idea to make this change....

**Comment by gonuke** [comment]
Thanks for the merge @pshriwise and bringing us formally into the new era of mcnp2cad!

Are there now some additional fixes that are waiting in other branches?  E.g. `fix_igeom_dep`

File: MCNPInput.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (pshriwise) [reply MCNPInput.hpp]: I'll make an issue as a reminder about this.

**Comment by pshriwise** [comment]
Yeah... I suppose there are. I'm kinda confused though. There is a fix for lattice translations in `fix_igeom_dep` for lattice translations that should've been included with the merge of #41. Looks like some git forensics are in order... :male_detective:

**Pull request merged**